--- a/source/Files.cpp
+++ b/source/Files.cpp
@@ -126,8 +126,10 @@
 		SDL_free(basePath);
 #endif
 
+#ifndef __3DS__
 		if(Exists(resources))
 			resources = filesystem::canonical(resources);
+#endif
 
 #if defined __linux__ || defined __FreeBSD__ || defined __DragonFly__
 		// Special case, for Linux: the resource files are not in the same place as
@@ -171,7 +173,9 @@
 	if(!Exists(config))
 		throw runtime_error("Unable to create config directory!");
 
+#ifndef __3DS__
 	config = filesystem::canonical(config);
+#endif
 
 	savePath = config / "saves";
 	CreateFolder(savePath);
